cpp/src/copying/gather.cu

```diff
@@ -221,24 +221,36 @@ struct column_gatherer {
   }
 };
 
+/**---------------------------------------------------------------------------*
+* @brief Function object for applying a transformation on the gathermap
+* that converts negative indices to positive indices
+*---------------------------------------------------------------------------**/
 template <typename MapType>
-struct index_map : public thrust::unary_function<MapType,MapType>
+struct map_transform : public thrust::unary_function<MapType,MapType>
 {
-  index_map(gdf_size_type source_n_rows) : source_n_rows(source_n_rows){}
+  map_transform(gdf_size_type n_rows, bool transform_negative_indices)
+    : n_rows(n_rows), transform_negative_indices(transform_negative_indices){}
```

Comment:
I think you should make `transform_negative_indices` a `bool` template parameter. Otherwise you pay the if/else cost on every index.

Reply:
Or have separate functors for transform_negative and default.

Reply:
Done.